WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/Coupon Text Validators/CouponAmountInputFormatter.swift

```diff
@@ -14,14 +14,16 @@ struct CouponAmountInputFormatter: UnitInputFormatter {
         guard input.isNotEmpty else {
             return false
         }
+
         return priceInputFormatter.isValid(input: input)
     }
 
     func format(input text: String?) -> String {
-        guard text.isNilOrEmpty else {
-            return priceInputFormatter.format(input: text)
+        guard !text.isNilOrEmpty else {
+            return "0"
         }
-        return "0"
+
+        return priceInputFormatter.format(input: text)
```

Comment:
Could you also please explain why we need these changes?

Reply:
We don't need it 😅 , It was just a side refactor when I noticed that the guard `else` was doing the actual function work stated as `format` and the function body return was the exception case, so I just switched the semantics to leave the format work into the function body. But I can just revert it since it's not related to the PR at all! Let me know what you think.

Reply:
This is fine, I was just asking in case I miss something here 😅